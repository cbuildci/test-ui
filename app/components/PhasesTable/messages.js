/*
 * PhasesTable Messages
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
    headerPhase: {
        id: 'app.components.PhasesTable.headerPhase',
        defaultMessage: 'Phase',
    },
    headerStatus: {
        id: 'app.components.PhasesTable.headerStatus',
        defaultMessage: 'Status',
    },
    headerDuration: {
        id: 'app.components.PhasesTable.headerDuration',
        defaultMessage: 'Duration',
    },
});
