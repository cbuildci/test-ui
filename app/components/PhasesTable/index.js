/**
 * PhasesTable
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import messages from './messages';
import TimeDuration from 'components/TimeDuration';
import PhaseStatus from '../PhaseStatus';
import { PHASE_TYPES } from '../../utils/constants';

function PhasesTable({ currentPhase, phases }) {
    const phaseMap = {};

    if (phases) {
        for (const phase of phases) {
            phaseMap[phase.phaseType] = phase;
        }
    }

    return (
        <table className="table table-md w-auto">
            <thead className="thead-light">
                <tr>
                    <th><FormattedMessage {...messages.headerPhase}/></th>
                    <th><FormattedMessage {...messages.headerStatus}/></th>
                    <th><FormattedMessage {...messages.headerDuration}/></th>
                </tr>
            </thead>
            <tbody>
                {PHASE_TYPES.map((phaseType) => {
                    const phase = phaseMap[phaseType];
                    let phaseStatus = null;

                    if (phase) {
                        phaseStatus = phase.phaseStatus;

                        if (phase.phaseType === 'COMPLETED') {
                            phaseStatus = 'SUCCEEDED';
                        }
                        else if (!phaseStatus && phase.phaseType === currentPhase) {
                            phaseStatus = 'IN_PROGRESS';
                        }
                    }

                    return (
                        <React.Fragment key={phaseType}>
                            <tr>
                                <th>{phaseType}</th>
                                <td>{phase ? <PhaseStatus status={phaseStatus}/> : '-'}</td>
                                <td>{phase && typeof phase.durationInSeconds === 'number' ? <TimeDuration seconds={phase.durationInSeconds}/> : '-'}</td>
                            </tr>
                            {phase && phase.contexts && phase.contexts.length > 0 && (
                                <React.Fragment>
                                    {phase.contexts.map((context, i) => (
                                        <tr key={i}>
                                            <td colSpan="3">{context.message}</td>
                                        </tr>
                                    ))}
                                </React.Fragment>
                            )}
                        </React.Fragment>
                    );
                })}
            </tbody>
        </table>
    );
}

PhasesTable.propTypes = {
    currentPhase: PropTypes.string,
    phases: PropTypes.arrayOf(PropTypes.object),
};

PhasesTable.defaultProps = {
    currentPhase: null,
    phases: null,
};

export default PhasesTable;
