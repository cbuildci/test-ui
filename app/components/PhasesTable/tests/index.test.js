import React from 'react';
import { shallow, render } from 'enzyme';
import { IntlProvider } from 'react-intl';
import PhasesTable from '../index';

const snapshots = (jsx) => {
    expect(shallow(jsx)).toMatchSnapshot();
    expect(render(
        <IntlProvider locale="en" initialNow={1500000100000}>{jsx}</IntlProvider>
    )).toMatchSnapshot();
};

describe('<PhasesTable />', () => {
    it('should render expected JSX', () => {
        snapshots(
            <PhasesTable
                phases={[
                    {
                        startTime: '2018-07-27T16:52:18.099Z',
                        endTime: '2018-07-27T16:52:18.496Z',
                        phaseStatus: 'SUCCEEDED',
                        durationInSeconds: 0,
                        phaseType: 'SUBMITTED',
                    },
                    {
                        startTime: '2018-07-27T16:52:18.496Z',
                        endTime: '2018-07-27T16:52:27.228Z',
                        contexts: [],
                        phaseStatus: 'SUCCEEDED',
                        durationInSeconds: 8,
                        phaseType: 'PROVISIONING',
                    },
                    {
                        startTime: '2018-07-27T16:52:27.228Z',
                        endTime: '2018-07-27T16:52:27.481Z',
                        contexts: [],
                        phaseStatus: 'SUCCEEDED',
                        durationInSeconds: 0,
                        phaseType: 'DOWNLOAD_SOURCE',
                    },
                    {
                        startTime: '2018-07-27T16:52:27.481Z',
                        endTime: '2018-07-27T16:52:32.601Z',
                        contexts: [],
                        phaseStatus: 'SUCCEEDED',
                        durationInSeconds: 5,
                        phaseType: 'INSTALL',
                    },
                    {
                        startTime: '2018-07-27T16:52:32.601Z',
                        endTime: '2018-07-27T16:52:32.677Z',
                        contexts: [],
                        phaseStatus: 'SUCCEEDED',
                        durationInSeconds: 0,
                        phaseType: 'PRE_BUILD',
                    },
                    {
                        startTime: '2018-07-27T16:52:32.677Z',
                        endTime: '2018-07-27T16:52:32.802Z',
                        contexts: [],
                        phaseStatus: 'SUCCEEDED',
                        durationInSeconds: 0,
                        phaseType: 'BUILD',
                    },
                    {
                        startTime: '2018-07-27T16:52:32.802Z',
                        endTime: '2018-07-27T16:52:32.904Z',
                        contexts: [],
                        phaseStatus: 'SUCCEEDED',
                        durationInSeconds: 0,
                        phaseType: 'POST_BUILD',
                    },
                    {
                        startTime: '2018-07-27T16:52:32.904Z',
                        endTime: '2018-07-27T16:52:32.983Z',
                        contexts: [],
                        phaseStatus: 'SUCCEEDED',
                        durationInSeconds: 0,
                        phaseType: 'UPLOAD_ARTIFACTS',
                    },
                    {
                        startTime: '2018-07-27T16:52:32.983Z',
                        endTime: '2018-07-27T16:52:35.099Z',
                        contexts: [],
                        phaseStatus: 'SUCCEEDED',
                        durationInSeconds: 2,
                        phaseType: 'FINALIZING',
                    },
                    {
                        phaseType: 'COMPLETED',
                        startTime: '2018-07-27T16:52:35.099Z',
                        endTime: null,
                    },
                ]}/>,
        );
    });
});
