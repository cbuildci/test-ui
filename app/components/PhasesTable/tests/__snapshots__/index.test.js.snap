// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhasesTable /> should render expected JSX 1`] = `
<table
  className="table table-md w-auto"
>
  <thead
    className="thead-light"
  >
    <tr>
      <th>
        <FormattedMessage
          defaultMessage="Phase"
          id="app.components.PhasesTable.headerPhase"
          values={Object {}}
        />
      </th>
      <th>
        <FormattedMessage
          defaultMessage="Status"
          id="app.components.PhasesTable.headerStatus"
          values={Object {}}
        />
      </th>
      <th>
        <FormattedMessage
          defaultMessage="Duration"
          id="app.components.PhasesTable.headerDuration"
          values={Object {}}
        />
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>
        SUBMITTED
      </th>
      <td>
        <PhaseStatus
          status="SUCCEEDED"
        />
      </td>
      <td>
        <TimeDuration
          seconds={0}
        />
      </td>
    </tr>
    <tr>
      <th>
        PROVISIONING
      </th>
      <td>
        <PhaseStatus
          status="SUCCEEDED"
        />
      </td>
      <td>
        <TimeDuration
          seconds={8}
        />
      </td>
    </tr>
    <tr>
      <th>
        DOWNLOAD_SOURCE
      </th>
      <td>
        <PhaseStatus
          status="SUCCEEDED"
        />
      </td>
      <td>
        <TimeDuration
          seconds={0}
        />
      </td>
    </tr>
    <tr>
      <th>
        INSTALL
      </th>
      <td>
        <PhaseStatus
          status="SUCCEEDED"
        />
      </td>
      <td>
        <TimeDuration
          seconds={5}
        />
      </td>
    </tr>
    <tr>
      <th>
        PRE_BUILD
      </th>
      <td>
        <PhaseStatus
          status="SUCCEEDED"
        />
      </td>
      <td>
        <TimeDuration
          seconds={0}
        />
      </td>
    </tr>
    <tr>
      <th>
        BUILD
      </th>
      <td>
        <PhaseStatus
          status="SUCCEEDED"
        />
      </td>
      <td>
        <TimeDuration
          seconds={0}
        />
      </td>
    </tr>
    <tr>
      <th>
        POST_BUILD
      </th>
      <td>
        <PhaseStatus
          status="SUCCEEDED"
        />
      </td>
      <td>
        <TimeDuration
          seconds={0}
        />
      </td>
    </tr>
    <tr>
      <th>
        UPLOAD_ARTIFACTS
      </th>
      <td>
        <PhaseStatus
          status="SUCCEEDED"
        />
      </td>
      <td>
        <TimeDuration
          seconds={0}
        />
      </td>
    </tr>
    <tr>
      <th>
        FINALIZING
      </th>
      <td>
        <PhaseStatus
          status="SUCCEEDED"
        />
      </td>
      <td>
        <TimeDuration
          seconds={2}
        />
      </td>
    </tr>
    <tr>
      <th>
        COMPLETED
      </th>
      <td>
        <PhaseStatus
          status="SUCCEEDED"
        />
      </td>
      <td>
        -
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<PhasesTable /> should render expected JSX 2`] = `
<table
  class="table table-md w-auto"
>
  <thead
    class="thead-light"
  >
    <tr>
      <th>
        <span>
          Phase
        </span>
      </th>
      <th>
        <span>
          Status
        </span>
      </th>
      <th>
        <span>
          Duration
        </span>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>
        SUBMITTED
      </th>
      <td>
        <span
          style="color:#28a745"
        >
          <i
            class="fas fa-check fa-fw mr-1"
          />
          <span>
            SUCCEEDED
          </span>
        </span>
      </td>
      <td>
        <span>
          0 seconds
        </span>
      </td>
    </tr>
    <tr>
      <th>
        PROVISIONING
      </th>
      <td>
        <span
          style="color:#28a745"
        >
          <i
            class="fas fa-check fa-fw mr-1"
          />
          <span>
            SUCCEEDED
          </span>
        </span>
      </td>
      <td>
        <span>
          8 seconds
        </span>
      </td>
    </tr>
    <tr>
      <th>
        DOWNLOAD_SOURCE
      </th>
      <td>
        <span
          style="color:#28a745"
        >
          <i
            class="fas fa-check fa-fw mr-1"
          />
          <span>
            SUCCEEDED
          </span>
        </span>
      </td>
      <td>
        <span>
          0 seconds
        </span>
      </td>
    </tr>
    <tr>
      <th>
        INSTALL
      </th>
      <td>
        <span
          style="color:#28a745"
        >
          <i
            class="fas fa-check fa-fw mr-1"
          />
          <span>
            SUCCEEDED
          </span>
        </span>
      </td>
      <td>
        <span>
          5 seconds
        </span>
      </td>
    </tr>
    <tr>
      <th>
        PRE_BUILD
      </th>
      <td>
        <span
          style="color:#28a745"
        >
          <i
            class="fas fa-check fa-fw mr-1"
          />
          <span>
            SUCCEEDED
          </span>
        </span>
      </td>
      <td>
        <span>
          0 seconds
        </span>
      </td>
    </tr>
    <tr>
      <th>
        BUILD
      </th>
      <td>
        <span
          style="color:#28a745"
        >
          <i
            class="fas fa-check fa-fw mr-1"
          />
          <span>
            SUCCEEDED
          </span>
        </span>
      </td>
      <td>
        <span>
          0 seconds
        </span>
      </td>
    </tr>
    <tr>
      <th>
        POST_BUILD
      </th>
      <td>
        <span
          style="color:#28a745"
        >
          <i
            class="fas fa-check fa-fw mr-1"
          />
          <span>
            SUCCEEDED
          </span>
        </span>
      </td>
      <td>
        <span>
          0 seconds
        </span>
      </td>
    </tr>
    <tr>
      <th>
        UPLOAD_ARTIFACTS
      </th>
      <td>
        <span
          style="color:#28a745"
        >
          <i
            class="fas fa-check fa-fw mr-1"
          />
          <span>
            SUCCEEDED
          </span>
        </span>
      </td>
      <td>
        <span>
          0 seconds
        </span>
      </td>
    </tr>
    <tr>
      <th>
        FINALIZING
      </th>
      <td>
        <span
          style="color:#28a745"
        >
          <i
            class="fas fa-check fa-fw mr-1"
          />
          <span>
            SUCCEEDED
          </span>
        </span>
      </td>
      <td>
        <span>
          2 seconds
        </span>
      </td>
    </tr>
    <tr>
      <th>
        COMPLETED
      </th>
      <td>
        <span
          style="color:#28a745"
        >
          <i
            class="fas fa-check fa-fw mr-1"
          />
          <span>
            SUCCEEDED
          </span>
        </span>
      </td>
      <td>
        -
      </td>
    </tr>
  </tbody>
</table>
`;
